### PR TITLE
Implement a LS API to the generate function call template

### PR DESCRIPTION
**Pull request by nipunayf**
## Purpose
This PR implements the generation of function call text when a user selects a function from the helper pane. It handles both already imported and not-yet-imported functions.

- **Function Call Generation:** Generates the correct function call text based on whether the function is currently available, imported, or available for import.
- **Import Handling:** Automatically adds necessary import statements to the shadowed project if a selected function is not already imported.
- **Request/Response:**
    - Request: `filepath`, `codedata`, and `kind` (current, import, available).
    - Response: A string `template` representing the function call template.

### CODE REVIEW

File: flow-model-generator/modules/flow-model-generator-ls-extension/src/main/java/io/ballerina/flowmodelgenerator/extension/response/FunctionCallTemplateResponse.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    private String template;
+
+    public void setTemplate(String template) {
+        this.template = template;
+    }
+
+    public String template() {
+        return template;
+    }
+}
```
Comment (NipunaRanasinghe) [inline flow-model-generator/modules/flow-model-generator-ls-extension/src/main/java/io/ballerina/flowmodelgenerator/extension/response/FunctionCallTemplateResponse.java:36]: Is there a reason why we can't use a record in here instead?

Comment (nipunayf) [reply flow-model-generator/modules/flow-model-generator-ls-extension/src/main/java/io/ballerina/flowmodelgenerator/extension/response/FunctionCallTemplateResponse.java:36]: We are extending the `AbstractFlowModelResponse`, but records don't support inheritance

Comment (NipunaRanasinghe) [reply flow-model-generator/modules/flow-model-generator-ls-extension/src/main/java/io/ballerina/flowmodelgenerator/extension/response/FunctionCallTemplateResponse.java:36]: didnt notice that this is an extended class. my bad

**Review by NipunaRanasinghe — DISMISSED** [review]
LGTM

**Review by NipunaRanasinghe — APPROVED** [review]

**Pull request merged**